invesalius/data/imagedata_utils.py

```diff
@@ -575,6 +576,46 @@ def convert_world_to_voxel(xyz, affine):
     return ijk
 
 
+# TODO: Add a description of different coordinate systems, including:
+#       - InVesalius's internal coordinate system,
+#       - the world coordinate system,
+#       - the image coordinate system.
+#
+def convert_invesalius_to_world(position, orientation):
+    """
+    Convert position and orientation from the Invesalius space to the world space.
+
+    Uses 'affine' matrix defined in the project. If it is undefined, return two Nones.
+
+    More information: https://nipy.org/nibabel/coordinate_systems.html
+
+    :param position: a vector of 3 coordinates in InVesalius space.
+    :param orientation: a vector of 3 Euler angles in InVesalius space.
+    :return: a tuple consisting of 3 coordinates and 3 Euler angles in world space, or two Nones if 'affine' matrix
+             is not defined in the project.
+    """
+    slice = sl.Slice()
+
+    affine = slice.affine
+    if slice.affine is None:
+        return None, None
+
+    world2inv = np.linalg.inv(affine)
+    world2inv[1, -1] -= slice.spacing[1] * slice.matrix.shape[1]
```

Comment:
This translation is likely to be wrong; it gives way too large y-coordinates after InVesalius->world transformation. I'll look into it, but for now, I'll close this patch and reopen it once fixed.